.github/workflows/release.yml

```diff
@@ -0,0 +1,118 @@
+name: Release - cli
+
+on:
+  workflow_dispatch:
+  pull_request:
+  schedule:
+    # 08:00am UTC everyday: https://crontab.guru/#0_8_*_*_*
+    # https://dateful.com/convert/utc?t=8am
+    - cron: "0 8 * * *"
+
+jobs:
+  lint-all:
+    if: |
+      github.event_name == 'schedule' ||
+      github.event_name == 'workflow_dispatch' ||
+      startsWith(github.head_ref, 'release/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: npm
+      - run: node bin/npm-cli.js run resetdeps
+      - run: node bin/npm-cli.js run lint-all --ignore-scripts
+
+  smoke-publish:
+    if: |
+      github.event_name == 'schedule' ||
+      github.event_name == 'workflow_dispatch' ||
+      startsWith(github.head_ref, 'release/')
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 12.13.0
+          - 12.x
+          - 14.15.0
+          - 14.x
+          - 16.0.0
+          - 16.x
+        platform:
+          - os: ubuntu-latest
+            shell: bash
+          - os: macos-latest
+            shell: bash
+        # XXX: this should be possible in windows also
+        # but resetdeps cant be a bash script
```

Comment:
This is _almost_ ready to run in Windows, but needs resetdeps and the `rm -rf` to be cross platform. I'm saving that for a future PR.